scrapy/core/scheduler.py

```diff
@@ -103,32 +145,54 @@ def _mqpush(self, request):
 
     def _dqpop(self):
         if self.dqs:
-            d = self.dqs.pop()
-            if d:
-                return request_from_dict(d, self.spider)
+            return self.dqs.pop()
+
+    def _mqpop(self):
+        return self.mqs.pop()
```

Comment:
What would you think about removing this method? It has a single, short line, and it is used only once.

You might even consider removing `_dqpop`, now that you’ve simplified it significantly.

Reply:
I would prefer to preserve `_dqpop`, as it reduces complexity of already `if`-reach code